00-programming-fundamentals/xx-debug.md

```diff
@@ -0,0 +1,69 @@
+#Bug Off!
+## Debugging with Pry
+
+
+Remember way back when we were showing off new gems we discovered from a very !? Well you've been using them ALL, right? At least Pry?
+
+
+### What is Pry
+There are two main uses for PRY, to use in place of IRB and to step through our code while running a program in terminal.
+
+Pry is another Ruby REPL, much like IRB, with some added functionality.
+
+When Instead of using a bunch of 'puts' commands in a program, we can use PRY to pause our code and
+
+
+
+### How to get started
+
+1. Install by running `` gem install pry `` in terminal.
+1. At the top of a file you want to debug, `` require 'pry' ``.
+1. In your code, write ``binding.pry`` anywhere in your code where you want to pause the code that's being run. Doing so will allow you to evaluate what code is doing up to that point, like checking what what a variable equals while being changed in a loop.
+
+
+
+### Test it out
+
+Because programs get read from top to bottom, any code that comes after a binding.pry command will not be read. Run the code in the exercise below, as you would normally. To exit, type 'exit' and push enter.
+
+``` Ruby
+require 'pry'
+def useful_method
```

Comment:
I think it would be useful to add the screenshot for what this looks like in the terminal when this breakpoint is hit